### ISSUE TITLE
Create a package for built-in functions

### ISSUE THREAD

**Issue by joeduffy** [pulumi/pulumi-terraform#267]
We decided to create a package, `@pulumi/terraform`, which contains all of the [Terraform built-in functions](https://www.terraform.io/docs/configuration/interpolation.html#built-in-functions) in raw, unopinionated form (same names, same semantics). This will help significantly with conversion tooling and efforts (for instance, `tf2pulumi` can use it).

**Comment by lukehoban** [comment]
Should we close out https://github.com/pulumi/pulumi/issues/1552 as a dupe of this?

**Comment by joeduffy** [comment]
Good progress on this one, moving out of M19 but keeping in M20 as I'm sure we'll keep chipping away.

**Comment by lukehoban** [comment]
@jen20 Should we make available the library as it currently stands and close this out and then track any remaining things to add as part of the new repo?

**Comment by jen20** [comment]
Yes, I think that's the best solution here - I'll do that today. The remaining functions either have very unusual semantics in Terraform, or are related to IPv6. I'll publish what we have, and open issues for each remaining function.

**Comment by g-price** [comment]
Is this still an active feature request? Really hoping for `cidrsubnet`. Thank you!

**Comment by lukehoban** [comment]
@jen20 Do we still have the work for this available somewhere?  Would love to just publish as an NPM package in current form.

**Comment by Tirke** [comment]
Is this still a thing ?

**Comment by lukehoban** [comment]
We haven't heard many requests/needs for this in the last 18 months, but in principle it's still "a good idea".  There was a PR that made a lot of progress on this at https://github.com/pulumi/pulumi-terraform/pull/268, but it was never finalized. Two things:
1. Do you have a particular need for any part of this?  Would love to hear more on your scenario/needs?
2. Definitely open to someone picking up the PR to finalize the work there if you are interested! :smile:

**Comment by dferretti** [comment]
:wave: I'll add one vote to the "interested" tally! We are using C# - if this were to happen I'm guessing it would need to be implemented separately in each language right? Since this isn't resource provider code there isn't any centralized golang implementation behind grpc or anything?

**Comment by shadiramadan** [comment]
Ah I was just looking for the equivalent CIDR helpers.  Sad to see they aren't available still!

**Comment by iwahbe** [comment]
We have effectively done this with [pulumi-std](https://github.com/pulumi/pulumi-std).
